test/export-api-def.test.js

```diff
@@ -102,6 +104,25 @@ describe('loopback:export-api-def generator', function() {
     });
   });
 
+  describe('running on project with onging async operations', function() {
+    var PROCESS_EXIT_STUB;
+
+    before(function stubProcessExit() {
+      PROCESS_EXIT_STUB = sinon.stub(process, 'exit');
+    });
+
+    it('does not hang', function() {
+      var gen = givenGenerator();
+      gen.run(function() {
+        expect(PROCESS_EXIT_STUB.called).to.be.equal(true);
```

Comment:
Interesting. How comes that the stub was already called, when there is a timeout of 30ms? Will it always be the case that the generator finishes only after the timeout was triggered, or is this introducing a possible false test failure when timing gets wrong? Maybe the test should periodically poll the value of `PROCESS_EXIT_STUB.called` and finish the test only after it changed the value to true? That would make the original test name "does not hang" sensible again.

Reply:
I realized that for some reason, the test was always passing because I had forgotten the done callback. I'm really not sure why the test passed although the `process.exit` was clearly not called at the time of the assertion. 

I've updated the code with your approach by polling every 10 milliseconds, until the test timeouts or `process.exit` gets called. Also added a test that checks `process.exit` is not called right away.

I've also fixed your other comments, let me know if the changes suit you.